assets/src/dashboard/app/views/myStories/header/index.js

```diff
@@ -107,33 +104,25 @@ function Header({
     return (
       <>
         {STORY_STATUSES.map((storyStatus) => {
-          const isStoryPrivate = storyStatus.status === STORY_STATUS.PRIVATE;
-          const cantReadPrivate =
-            !canReadPrivatePosts ||
-            !totalStoriesByStatus.private ||
-            totalStoriesByStatus.private < 1;
-
-          if (isStoryPrivate && cantReadPrivate) {
+          const { label, status, value } = storyStatus;
+          if (!(status in totalStoriesByStatus)) {
             return null;
           }
-          const label = storyStatus.label;
-          const labelCount = totalStoriesByStatus?.[storyStatus.status] ? (
-            <span>{totalStoriesByStatus?.[storyStatus.status]}</span>
+          const labelCount = totalStoriesByStatus?.[status] ? (
+            <span>{totalStoriesByStatus?.[status]}</span>
           ) : null;
```

Comment:
Because of the `status in totalStoriesByStatus` check above this will never be `null`.

So can just simplify to

```js
const labelCount = <span>{totalStoriesByStatus?.[status]}</span>
```

That also means we don't need the ` {labelCount && labelCount}` check below.

Which means we can just write:

```jsx
<StyledPill ...>
{label}
<span>{totalStoriesByStatus.[status]}</span>
</StyledPill>

Reply:
So the doesn't work, because you should not render a 0 if there is one. I have change the logic a little make it a little cleaner.